pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -0,0 +1,333 @@
+}
+
+// Process implements the Processor interface (so required to be public)
+func (c *Controller) Process(ctx context.Context, key, ns, name string) autoscaling.ProcessResult {
+	res, err := c.processPodAutoscaler(ctx, key, ns, name)
+	if err != nil {
+		numRequeues := c.Workqueue.NumRequeues(key)
+		log.Errorf("Impossible to synchronize DatadogPodAutoscaler (attempt #%d): %s, err: %v", numRequeues, key, err)
+
+		if numRequeues >= maxRetry {
+			log.Infof("Max retries reached for DatadogPodAutoscaler: %s, removing from queue", key)
+			res = autoscaling.NoRequeue
+		}
+	}
+
+	return res
+}
+
+func (c *Controller) processPodAutoscaler(ctx context.Context, key, ns, name string) (autoscaling.ProcessResult, error) {
+	podAutoscaler := &datadoghq.DatadogPodAutoscaler{}
+	podAutoscalerCachedObj, err := c.Lister.ByNamespace(ns).Get(name)
+	if err == nil {
+		err = autoscaling.FromUnstructured(podAutoscalerCachedObj, podAutoscaler)
+	}
+
+	switch {
+	case errors.IsNotFound(err):
+		// We ignore not found here as we may need to create a DatadogPodAutoscaler later
+		podAutoscaler = nil
+	case err != nil:
+		return autoscaling.Requeue, fmt.Errorf("Unable to retrieve DatadogPodAutoscaler: %w", err)
+	case podAutoscalerCachedObj == nil:
+		return autoscaling.Requeue, fmt.Errorf("Could not parse empty DatadogPodAutoscaler from local cache")
+	}
+
+	// No error path, check what to do with this event
+	if c.IsLeader() {
+		return c.syncPodAutoscaler(ctx, key, ns, name, podAutoscaler)
+	}
+
+	// In follower mode, we simply sync updates from Kubernetes
+	// If the object is present in Kubernetes, we will update our local version
+	// Otherwise, we clear it from our local store
+	if podAutoscaler != nil {
+		c.store.Set(key, model.NewPodAutoscalerInternal(key, podAutoscaler), c.ID)
+	} else {
+		c.store.Delete(key, c.ID)
+	}
+
+	return autoscaling.NoRequeue, nil
+}
+
+// Synchronize DatadogPodAutoscaler state between internal store and Kubernetes objects
+// Make sure any `return` has the proper store Unlock
+func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string, podAutoscaler *datadoghq.DatadogPodAutoscaler) (autoscaling.ProcessResult, error) {
+	podAutoscalerInternal, podAutoscalerInternalFound := c.store.LockRead(key, true)
+
+	// Object is missing from our store
+	if !podAutoscalerInternalFound {
+		if podAutoscaler != nil {
+			// If we don't have an instance locally, we create it. Deletion is handled through setting the `Deleted` flag
+			c.store.UnlockSet(key, model.NewPodAutoscalerInternal(key, podAutoscaler), c.ID)
+		} else {
+			// If podAutoscaler == nil, both objects are nil, nothing to do
+			c.store.Unlock(key)
+		}
+
+		return autoscaling.NoRequeue, nil
+	}
+
+	if podAutoscaler == nil {
+		// Object is not present in Kubernetes
+		// If flagged for deletetion, we just need to clear up our store (deletion complete)
+		// Also if object was not owned by remote config, we also need to delete it (deleted by user)
```

Comment:
```suggestion
		// Also if internal object was not owned by remote config, we also need to delete it (deleted by user)
```
nit

Reply:
Meaning is intended, object and internal object are synced.